batchcloser/closers.go

```diff
@@ -49,7 +49,7 @@ type BatchCloser struct {
 func (bc *BatchCloser) Close() error {
 	var errs errorsbp.Batch
 	for _, closer := range bc.closers {
-		errs.AddPrefix(fmt.Sprintf("%#v", closer), closer.Close())
+		errs.AddPrefix(fmt.Sprintf("%T", closer), closer.Close())
```

Comment:
🔕 (this is a bit out of scope, and it is a somewhat major reduction in the verbosity of the prefix, might be worth not doing this in this PR at least)

Reply:
Not doing this will cause a race condition in the `timestampCloser` used in tests, which seems to be a new issue in go 1.19 (we don't see this race condition from before)

I'll add a note in the PR description/commit message.

Reply:
ok I reverted this and implemented `GoString` on `timestampCloser` instead.